edx/edx-app-ios
# iOS - Update subsection listing and component view for gated content

### Description

[LEARNER-6654](https://openedx.atlassian.net/browse/LEARNER-6654)

In this PR the subsection list and component view is updated for gated content, the blocks having gated content will now show message of not availability in list view and by opening they will show the full screen error message with lock icon.

Open source community have different versions of edX server, To support different versions of API URL we made them configurable. If the API_URL_VERSION is not mention in configuration the default version will be use(inside the app's source code). 
Currently we have added config variable for blocks API with version "v2", in case the config would not update the default version "v1" will be use.

Detail can also be found on [App Configuration Flags](https://openedx.atlassian.net/wiki/spaces/LEARNER/pages/48792067/App+Configuration+Flags)
```
API_URL_VERSION: 
         BLOCKS: 'v2'

```
### How to test this PR
- Login with testing account 
   username: rtester@yopmail.com
   password: Fedx1234
- Add above mention configurations in config file.
- Go in course detail view there is multiple subsections having blocks with gated content. 
   Sample subsections: "Fronts and Air Masses", "Weather Prediction Assignment Instructions"

- [ ] Gated block should have disable state icon.
- [ ] Gated block should have subtitle(Not available on Mobile) in list view.
- [ ] By tapping and on gated block it should show full screen error message
